### PR TITLE
[ACA-4421] [Edit Aspects] Apply button should be disabled by default

### PR DESCRIPTION
**Pull request by dhrn**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/ACA-4421

**What is the new behaviour?**
fixed

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/aspect-list/aspect-list.component.ts [hunk 1]
```diff
@@ -57,13 +57,9 @@ export class AspectListComponent implements OnInit, OnDestroy {
     ngOnInit(): void {
         if (this.nodeId) {
             const node$ = this.nodeApiService.getNode(this.nodeId);
-            const customAspect$ = this.aspectListService.getCustomAspects()
-            .pipe(map(
```
Comment (VitoAlbano) [inline lib/content-services/src/lib/aspect-list/aspect-list.component.ts:61]: Why have you removed this part? the aspect needs to be filtered when showed up and only the one in the config should be showed.

**Review by VitoAlbano — CHANGES_REQUESTED** [review]

File: lib/content-services/src/lib/aspect-list/aspect-list.component.ts [hunk 2]
```diff
@@ -109,4 +105,11 @@ export class AspectListComponent implements OnInit, OnDestroy {
     getTitle(aspect: any): string {
         return aspect?.entry?.title ? aspect?.entry?.title : aspect?.entry?.id;
     }
+
+    hasEqualAspect(): boolean {
```
Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list.component.ts:109]: Can we  use .includes insted of .find and have something like this:

`return this.nodeAspectStatus?.length !== this.nodeAspects.length ? false : this.nodeAspectStatus.every((aspect) => this.nodeAspects.includes(element);`

Even better wrap the last part into a function

`return this.nodeAspectStatus?.length !== this.nodeAspects.length ? false : isEveryAspectEtcEtc()`


Also can we rename this method to hasModifiedAspects, hasUpdatedAspects or something similar?

File: lib/content-services/src/lib/aspect-list/aspect-list.component.ts [hunk 3]
```diff
@@ -57,13 +57,9 @@ export class AspectListComponent implements OnInit, OnDestroy {
...
             const node$ = this.nodeApiService.getNode(this.nodeId);
-            const customAspect$ = this.aspectListService.getCustomAspects()
-            .pipe(map(
-                (customAspects) => customAspects.flatMap((customAspect) => customAspect.entry.id)
-            ));
-            this.aspects$ = zip(node$, customAspect$).pipe(
-                tap(([node, customAspects]) => {
-                    this.nodeAspects = node.aspectNames.filter((aspect) => this.aspectListService.getVisibleAspects().includes(aspect) || customAspects.includes(aspect));
+            this.aspects$ = node$.pipe(
+                tap((node) => {
```
Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list.component.ts:61]: can we add the type of the node? We should have such classes from what I remember

File: lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts [hunk 4]
```diff
@@ -303,6 +303,17 @@ describe('AspectListDialogComponent', () => {
             expect(customAspectCheckbox).not.toBeNull();
             expect(customAspectCheckbox.checked).toBeTruthy();
         });
+
+        it('Should apply button disabled till aspect-list updated', async () => {
+            await fixture.detectChanges();
+            expect(fixture.nativeElement.querySelector('#aspect-list-dialog-actions-apply')).toBeDefined();
+            expect(fixture.nativeElement.querySelector('#aspect-list-dialog-actions-apply').disabled).toBe(true);
```
Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts:310]: can we store this into a variable and use it in the 2 expectations?

File: lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts [hunk 5]
```diff
@@ -303,6 +303,17 @@ describe('AspectListDialogComponent', () => {
...
+
+            fixture.nativeElement.querySelector('#aspect-list-dialog-actions-clear').click();
+
+            await fixture.detectChanges();
+            expect(fixture.nativeElement.querySelector('#aspect-list-dialog-actions-apply').disabled).toBe(false);
```
Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts:315]: can we store every element we query into a variable? In this way we can debug them easier in the future

Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts:315]: Can we split it to 2 tests as this is checking more than one things?

* Should apply button be disabled by default
* Should apply button get enabled when the aspect list gets updated

File: lib/content-services/src/lib/aspect-list/aspect-list.component.spec.ts [hunk 6]
```diff
@@ -156,6 +156,12 @@ describe('AspectListComponent', () => {
             fixture.destroy();
         });
 
+        it('should return true when same aspect list selected', () => {
+            expect(component.hasEqualAspect()).toBe(true);
+            component.clear();
+            expect(component.hasEqualAspect()).toBe(false);
```
Comment (arditdomi) [inline lib/content-services/src/lib/aspect-list/aspect-list.component.spec.ts:162]: Can we split to 2? 

* Should return true when etc...
* Should return false when etc...

**Review by arditdomi — CHANGES_REQUESTED** [review]

**Review by VitoAlbano — APPROVED** [review]

**Review by eromano — APPROVED** [review]

**Review by arditdomi — APPROVED** [review]

**Pull request merged**